example/lib/peripheral_details/peripheral_details_bloc.dart

```diff
@@ -1,22 +1,41 @@
 import 'dart:async';
 import 'package:blemulator_example/adapter/ble_adapter.dart';
-import 'package:blemulator_example/model/ble_peripheral.dart';
+import 'package:blemulator_example/peripheral_details/peripheral_info.dart';
 import 'package:bloc/bloc.dart';
 import './bloc.dart';
 
 class PeripheralDetailsBloc
     extends Bloc<PeripheralDetailsEvent, PeripheralDetailsState> {
   BleAdapter _bleAdapter;
-  final BlePeripheral _chosenPeripheral;
+  final String _peripheralIdentifier;
 
-  PeripheralDetailsBloc(this._bleAdapter, this._chosenPeripheral);
+  PeripheralDetailsBloc(this._bleAdapter, this._peripheralIdentifier);
 
   @override
-  PeripheralDetailsState get initialState =>
-      PeripheralDetailsState(peripheral: _chosenPeripheral);
+  PeripheralDetailsState get initialState {
+    return _refreshStateFromAdapter();
```

Comment:
I'm not a fan of this. Perhaps BLoC should get name as well as a parameter and display only those two in the initial state and fetch any additional info later?